### PR TITLE
V3 java-SDK - ppplmer-96596 update query parameters - Part C

### PR DESCRIPTION
**Pull request by Rrathinasabapath**
Made query parameter changes for the following objects and their list methods. Please review and approve.

1. Transfers
2. Spendback
3. Balances
    - List User Balances
    - List Prepaid Card Balances
    - List Account Balances
 4. Receipts
    - List User Receipts
    - List Prepaid Card Receipts
    - List Account Receipts
 5. Webhook Notifications
6. Status Transitions
    - Bank Account
    - Bank Card
    - Paper Check
    - Payment
    - Paypal Acount
    - Prepaid Card
    - User
    - Venmo Account

### CODE REVIEW

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 1]
```diff
@@ -487,8 +487,11 @@ public void testListUserStatisTransitions_withParameters() throws Exception {
         Hyperwallet client = new Hyperwallet("test-username", "test-password");
         HyperwalletApiClient mockApiClient = createAndInjectHyperwalletApiClientMock(client);
 
-        HyperwalletPaginationOptions options = new HyperwalletPaginationOptions();
+        //HyperwalletPaginationOptions options = new HyperwalletPaginationOptions();
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:490]: Pls remove this commented line.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccount.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
     private Type type;
     private Status status;
+    private Status transition;
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccount.java:24]: Bank Account transition has only two possible states:
1. DE_ACTIVATED
2. ACTIVATED
Please create a separate enum to fix this.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccountListPaginationOptions.java [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
     private HyperwalletBankAccount.Type type;
     private HyperwalletBankAccount.Status status;
+    private HyperwalletBankAccount.Status transition;
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccountListPaginationOptions.java:7]: Same comment as above.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankCard.java [hunk 4]
```diff
@@ -21,6 +21,7 @@
     
     private Type type;
     private Status status;
+    private Status transition;
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankCard.java:24]: Same comment as above.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletPaperCheck.java [hunk 5]
```diff
@@ -19,6 +19,8 @@
     private HyperwalletTransferMethod.Type type;
     private String token;
     private HyperwalletTransferMethod.Status status;
+    private HyperwalletTransferMethod.Status transition;
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletPaperCheck.java:22]: The comment with respect to transition is generic. Pls fix in all the places.

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPaymentTest.java [hunk 6]
```diff
@@ -10,14 +10,13 @@ protected HyperwalletPayment createBaseModel() {
         HyperwalletPayment payment = new HyperwalletPayment();
         payment
                 .status("COMPLETED")
+                .transition("COMPLETED")
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPaymentTest.java:13]: CANCELLED is the only available transition for HyperwalletPayment.

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPrepaidCardTest.java [hunk 7]
```diff
@@ -11,8 +11,8 @@ protected HyperwalletPrepaidCard createBaseModel() {
         prepaidCard
                 .token("test-token")
                 .type(HyperwalletTransferMethod.Type.PREPAID_CARD)
-
                 .status(HyperwalletTransferMethod.Status.ACTIVATED)
+                .transition(HyperwalletTransferMethod.Status.ACTIVATED)
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPrepaidCardTest.java:15]: Transition status is different for different objects. Please check the document and update.

**Review by ramahalingam — CHANGES_REQUESTED** [review]
Please fix the comments.

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java (hunk 1 above)
Comment (Rrathinasabapath) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:490]: Done

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPaymentTest.java (hunk 6 above)
Comment (Rrathinasabapath) [reply src/test/java/com/hyperwallet/clientsdk/model/HyperwalletPaymentTest.java:13]: Following are the status transitions allowed for HyperwalletPayment. 'COMPLETED' is one of them and 'CANCELLED' is not one of them. Details are per https://docs.hyperwallet.com/content/api/v3/resources/status-transitions/list. Please let me know if there is any error in my statement.

CREATED
SCHEDULED
PENDING_ACCOUNT_ACTIVATION
PENDING_ID_VERIFICATION
PENDING_TAX_VERIFICATION
PENDING_TRANSFER_METHOD_ACTION
PENDING_TRANSACTION_VERIFICATION
IN_PROGRESS
UNCLAIMED
COMPLETED
FAILED
RECALLED
RETURNED
EXPIRED

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletPaperCheck.java (hunk 5 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletPaperCheck.java:22]: These were not refactored to keep it in accordance with V4 and the existing approach taken to refer to HyperwalletTransferMethod class for other fields like type and status. Changing the approach will call for changes in V4 and also other methods where HyperwalletPaperCheck is called.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccount.java (hunk 2 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccount.java:24]: Per the documentation possible status' are as given below.

ACTIVATED
VERIFIED
INVALID
DE_ACTIVATED

and for transitions they are:

ACTIVATED
VERIFIED
INVALID
DE_ACTIVATED

They are declared in the following lines as enumerations already. Status is used for transition too.
    public static enum Status {ACTIVATED, VERIFIED, INVALID, DE_ACTIVATED}

Let me know if anymore change is still required.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccountListPaginationOptions.java (hunk 3 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccountListPaginationOptions.java:7]: Same comment as given above.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankCard.java (hunk 4 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankCard.java:24]: Same comment as given above.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccount.java (hunk 2 above)
Comment (ramahalingam) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccount.java:24]: I'm referring to this link: https://docs.hyperwallet.com/content/api/v3/resources/status-transitions/bank-accounts-create for Bank Account Status transition. Please let me know if I have to refer to another link.

Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBankAccount.java:24]: This is an inconsistency. A new ticket has to be created for this to update the document. However, addition of Verified and Invalid in the query will not break the functionality. Please approve the request now as I am going by the document https://docs.hyperwallet.com/content/api/v4/resources/status-transitions/bank-accounts-retrieve for now.

**Review by ramahalingam — APPROVED** [review]

**Review by vkorotchenko — APPROVED** [review]

**Review by arao6 — APPROVED** [review]

**Pull request merged**